### PR TITLE
Spread

### LINKED ISSUE #1476: Wire up spread aggregate function

### PR DESCRIPTION
**Pull request by corylanou**
Fixes #1476

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -2154,10 +2154,11 @@ func (s *Server) executeShowMeasurementsStatement(stmt *influxql.ShowMeasurement
 		row.Values = append(row.Values, []interface{}{v})
 	}
 
-	// Make a result.
+	// Make result.
 	result := &Result{
-		Rows: influxql.Rows{row},
+		Rows: make(influxql.Rows, 1),
 	}
+	result.Rows[0] = row
```
Comment (corylanou) [inline server.go]: This change was due to go vet failing.  There is an unfortunate side affect of typed slices outside of the package that it doesn't understand, and will likely not change this behavior.

**Comment by otoolep** [comment]
If the goal is to have each mapper send its local max and mix, and then the reducer emit the difference between the largest of all maxes and the smallest of all mins, then +1. I have one 1 minor comment.

File: influxql/engine.go [hunk 2]
```diff
@@ -753,6 +755,53 @@ func ReduceMax(key Key, values []interface{}, e *Emitter) {
...
+		// Initialize
+		if !pointsYielded {
+			result.max = val.max
+			result.min = val.min
+			pointsYielded = true
+		}
+		result.max = math.Max(result.max, val.max)
+		result.min = math.Min(result.min, val.min)
+	}
+	spread := result.max - result.min
```
Comment (otoolep) [inline influxql/engine.go]: Not much point calculating `spread` if `pointsYielded` is false after the `for` loop.

File: influxql/engine.go [hunk 3]
```diff
@@ -753,6 +755,53 @@ func ReduceMax(key Key, values []interface{}, e *Emitter) {
...
+	if pointsYielded {
+		e.Emit(key, spread)
```
Comment (otoolep) [inline influxql/engine.go]: `e.Emit(key, result.max - result.min)`

Comment (corylanou) [inline influxql/engine.go]: makes sense.  changed.

File: influxql/engine.go [hunk 4]
```diff
@@ -753,6 +755,52 @@ func ReduceMax(key Key, values []interface{}, e *Emitter) {
 	}
 }
 
+type spreadMapOutput struct {
+	min, max float64
```
Comment (pauldix) [inline influxql/engine.go]: These should be exported so that they can be serialized later

**Comment by pauldix** [comment]
I know you just merged, but I had one comment.

**Pull request merged**